Mathlib/Algebra/Module/ZLattice/Covolume.lean

```diff
@@ -130,9 +134,52 @@ theorem covolume_eq_det_inv {ι : Type*} [Fintype ι] (L : Submodule ℤ (ι →
     IsUnit.unit_spec, ← Basis.det_basis, LinearEquiv.coe_det]
   rfl
 
-theorem volume_image_eq_volume_div_covolume {ι : Type*} [Fintype ι]
-    (L : Submodule ℤ (ι → ℝ)) [DiscreteTopology L] [IsZLattice ℝ L] (b : Basis ι ℤ L)
-    {s : Set (ι → ℝ)} :
+/--
+Let `L₁` be a sub-`ℤ`-lattice of `L₂`. Then the index of `L₁` inside `L₂` is equal to
+`covolume L₁ / covolume L₂`.
+-/
+theorem covolume_div_covolume_eq_relindex {ι : Type*} [Fintype ι] [DecidableEq ι]
+    (L₁ L₂ : Submodule ℤ (ι → ℝ)) [DiscreteTopology L₁] [IsZLattice ℝ L₁] [DiscreteTopology L₂]
+    [IsZLattice ℝ L₂] (h : L₁ ≤ L₂) :
+    covolume L₁ / covolume L₂ = L₁.toAddSubgroup.relindex L₂.toAddSubgroup := by
+  let b₁ : Basis ι ℤ L₁ := (Free.chooseBasis ℤ L₁).reindex (Fintype.equivOfCardEq
+    (by rw [← finrank_eq_card_chooseBasisIndex, ZLattice.rank ℝ, finrank_fintype_fun_eq_card]))
+  let b₂ : Basis ι ℤ L₂ := (Free.chooseBasis ℤ L₂).reindex (Fintype.equivOfCardEq
+    (by rw [← finrank_eq_card_chooseBasisIndex, ZLattice.rank ℝ, finrank_fintype_fun_eq_card]))
```

Comment:
Is is possible to extract API for this? (choose a basis of a lattice, and describe its properties)